src/index.ts

```diff
@@ -1,74 +1,45 @@
 import {
-  OAuth2Profile,
   OAuth2Strategy,
   OAuth2StrategyVerifyParams,
 } from 'remix-auth-oauth2';
 import type { StrategyVerifyCallback } from 'remix-auth';
 
-import type { FacebookPicture } from './types';
+import type { AdditionalFacebookProfileField, FacebookProfile, FacebookScope, FacebookExtraParams, FacebookStrategyOptions } from './types';
+export * from './types';
 
-/**
- * This interface declares what configuration the strategy needs from the
- * developer to correctly work.
- */
-export interface FacebookStrategyOptions {
-  clientID: string;
-  clientSecret: string;
-  callbackURL: string;
-  scope?: string;
-}
-
-export type FacebookProfile = {
-  id: string;
-  displayName: string;
-  name: {
-    familyName: string;
-    givenName: string;
-  };
-  emails: [{ value: string }];
-  photos: [{ value: string }];
-  _json: {
-    id: string;
-    name: string;
-    first_name: string;
-    last_name: string;
-    picture: FacebookPicture;
-    email: string;
-  };
-} & OAuth2Profile;
+export const baseProfileFields = [
+	'id',
+	'email',
+	'name',
+	'first_name',
+	'middle_name',
+	'last_name',
+	'picture',
+] as const;
 
-/**
- * This interface declares what the developer will receive from the strategy
- * to verify the user identity in their system.
- */
-export type FacebookExtraParams = {
-  // FIXME!
-} & Record<string, string | number>;
+export const FacebookName = 'facebook';
+export const FacebookDefaultScopes: FacebookScope[] = ['public_profile', 'email'];
+export const FacebookScopeSeperator = ',';
```

Comment:
It's better for the user to define scopes as a typed array ['scope1', 'scope2'] and then we do the work to format that into facebook's preferred style